src/components/user-group/uc/deleteUserData.uc.js

```diff
@@ -0,0 +1,37 @@
+const { ValidationError } = require('../../../errors');
+const { classesRepo } = require('../repo/index');
+const { isValid: isValidObjectId } = require('../../../helper/compare').ObjectId;
+const { debug } = require('../../../logger');
+
+const validateParams = (userId) => {
+	if (!isValidObjectId(userId)) throw new ValidationError('a valid objectId is required', { userId });
+};
+
+const addClassesToTrashbinData = (classes = [], data) => {
+	const student = classes.filter((classItem) => classItem.student).map((classItem) => classItem._id);
+	const teacher = classes.filter((classItem) => classItem.teacher).map((classItem) => classItem._id);
+	Object.assign(data, { classIds: { student, teacher } });
+};
+
+const deleteUserDataFromClasses = async (userId) => {
+	validateParams(userId);
+	debug(`deleting user mentions in classes contents started`, { userId });
```

Comment:
I am not sure if we put this debug information at this point. Maybe a more generic way ..i want to execute on usecase with name and parameter and i had executed a usecase with parameter and get this result. If we go this way, we can try if we can put it around the uc in the same way for every uc.
Maybe by using a uc (interface) class.